domain/src/main/java/org/oppia/domain/util/StateRetriever.kt

```diff
@@ -477,4 +488,54 @@ class StateRetriever @Inject constructor(
     }
     return StringList.newBuilder().addAllHtml(list).build()
   }
+
+  private fun parseImageWithRegions(jsonObject: JSONObject): ImageWithRegions {
+    return ImageWithRegions.newBuilder()
+      .addAllLabelRegions(parseJsonToLabeledRegionsList(jsonObject.getJSONArray("labeledRegions")))
+      .setImagePath(jsonObject.getString("imagePath"))
+      .build()
+  }
+
+  private fun parseJsonToLabeledRegionsList(jsonArray: JSONArray): List<LabeledRegion> {
+    val list: MutableList<LabeledRegion> = ArrayList()
+    for (i in 0 until jsonArray.length()) {
+      list.add(parseLabeledRegion(jsonArray.getJSONObject(i)))
+    }
+    return list
+  }
+
+  private fun parseLabeledRegion(jsonObject: JSONObject): LabeledRegion {
+    return LabeledRegion.newBuilder()
+      .setLabel(jsonObject.getString("label"))
+      .setRegion(parseRegion(jsonObject.getJSONObject("region")))
+      .setRegionDescription(jsonObject.getString("regionDescription"))
+      .build()
+  }
+
+  private fun parseRegion(jsonObject: JSONObject): LabeledRegion.Region {
+    val regionType =
+      if (jsonObject.get("regionType") == LabeledRegion.Region.RegionType.RECTANGLE.name) {
```

Comment:
Could maybe be simplified to a ``when`` since we may want to add future cases here, e.g.:

```kotlin
private fun parseRegion(jsonObject: JSONObject): LabeledRegion.Region {
  return LabeledRegion.Region.newBuilder()
    .setRegionType(parseRegionType(jsonObject.get("regionType"))
    ...
    .build()
}

private fun parseRegionType(regionTypeStr: String): LabeledRegion.Region.RegionType {
  return when (regionTypeStr) {
    LabeledRegion.Region.RegionType.RECTANGLE.name -> LabeledRegion.Region.RegionType.RECTANGLE
    else -> LabeledRegion.Region.RegionType.UNRECOGNIZED
  }
}
```

Suggest also maybe importing Region or RegionType directly to shorten some of this new code.

Reply:
done